packages/flutter_tools/test/general.shard/android/android_project_migration_test.dart

```diff
@@ -260,14 +333,86 @@ tasks.register("clean", Delete) {
         expect(bufferLogger.traceText, contains(optOutFlagEnabled));
       });
     });
+
+    group('migrate min sdk versions less than 19 to flutter.minSdkVersion '
+        'when in a FlutterProject that is an app', ()
+    {
+      late MemoryFileSystem memoryFileSystem;
+      late BufferLogger bufferLogger;
+      late FakeAndroidProject project;
+      late MinSdkVersionMigration migration;
+
+      setUp(() {
+        memoryFileSystem = MemoryFileSystem.test();
+        memoryFileSystem.currentDirectory.childDirectory('android').createSync();
+        bufferLogger = BufferLogger.test();
+        project = FakeAndroidProject(
+          root: memoryFileSystem.currentDirectory.childDirectory('android'),
+        );
+        project.appGradleFile.parent.createSync(recursive: true);
+        migration = MinSdkVersionMigration(
+            project,
+            bufferLogger
+        );
+      });
+
+      testWithoutContext('do nothing when files missing test', () {
+        migration.migrate();
+        expect(bufferLogger.traceText, contains(appGradleNotFoundWarning));
+      });
+
+      testWithoutContext('replace when api 16 test', () {
+        project.appGradleFile.writeAsStringSync(sampleModuleGradleBuildFile(minSdk16));
```

Comment:
I think it'd make the test a bit better if the test's own source had this string explicitly, like
```suggestion
        const String minSdkVersionString = 'minSdkVersion 16';
        project.appGradleFile.writeAsStringSync(sampleModuleGradleBuildFile(minSdkVersionString));
```

That makes it so that one can pretty well read the test on its own and understand what scenario it's testing. With the version in the current branch, the reader can see what scenario it's *intended* to test — but in order to confirm that there isn't some mistake in that `minSdk16` string, one has to refer back to the same code that's being tested.

It also makes the test more independent of the implementation internals of the code it's testing, which is good for leaving the code free to be refactored.

Having the string in the test code is the way the [corresponding tests work](https://github.com/flutter/flutter/blob/7b47ba1ef92c7921fb28642488a20a6b73fc4164/packages/flutter_tools/test/general.shard/ios/ios_project_migration_test.dart#L551-L554) for that iOS minimum-version migration.

Reply:
Changed!